### ISSUE TITLE
Support rewriting source files when working directories differ

### ISSUE THREAD

**Issue by ayrtonm** [immunant/IA2-Phase2#210]
There seems to be a bug in clang where the rewriter API cannot write changes to disk if the working directories (as listed in the compile_commands.json) of any source files that are passed as input differ from the directory where the rewriter was invoked. Nginx's perl module auto-generates some source file that is compiled from a different working directory which is why we switched to the nginx rtmp module.

**Comment by ayrtonm** [comment]
May be related to https://github.com/llvm/llvm-project/issues/22759

**Comment by rinon** [comment]
Is this still an issue?

**Comment by ayrtonm** [comment]
yeah it's still an issue which we should fix. I don't think it should affect the plan to create intermediate source files though.
